### PR TITLE
Builtin CFITSIO: 3.45 -> 4.4.0 to avoid indirect CURL dependency

### PR DESCRIPTION
**Pull request by guitargeek**
This avoids the indirect dependence of ROOT on CURL, because the CURL dependence of CFITSIO is optional since version 3.49: https://github.com/HEASARC/cfitsio/blob/f220e6e2c570f19228609ee081f735df0ddb204b/docs/changes.txt#L199

Note: I tested the `builtin_cfitsio` locally on Linux, and for macOS and Linux the CI will cover it because these platforms configure with `builtin_cfitsio=ON`.

### CODE REVIEW

**Review by pcanal — APPROVED** [review]
LGTM.  Thanks.

**Review by pcanal — CHANGES_REQUESTED** [review]
The incremental build might need a refresh to properly see the new cftios header files.

**Comment by guitargeek** [comment]
> The incremental build might need a refresh to properly see the new cftios header files.

Good point! I asked the CI to do a clean build, and now it worked!

**Comment by guitargeek** [comment]
Argh, clicked the wrong button again. Please GitHub, increase the space between "Close with comment" and "Comment" :laughing:

**Review by couet — APPROVED** [review]

**Review by bellenot — CHANGES_REQUESTED** [review]
The build type is missing for Windows, then the default will be used, and that might break the debug builds (`CMAKE_BUILD_TYPE` is not used on Windows).
I.e. This check is missing:
```
if(WIN32)
  if(winrtdebug)
    set(cfitsiobuild "Debug")
  else()
    set(cfitsiobuild "Release")
  endif()
endif()

```
and then the build command (for Windows only I think):
```
        BUILD_COMMAND ${CMAKE_COMMAND} --build . --config ${cfitsiobuild}
```

**Comment by guitargeek** [comment]
Doesn't have setting the build type the same effect, which is what I do right now?
```
  CMAKE_ARGS -G ${CMAKE_GENERATOR}
             -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
```
I wonder if we can make the code more platform in-depended like this

**Comment by guitargeek** [comment]
@bellenot, the zlib problem is fixed now also on Windows, but there are sill build errors related to something else:
```
      (compiling source file '../BUILTIN_CFITSIO/eval_y.c')
      
  C:\ROOT-CI\build\CFITSIO-prefix\src\BUILTIN_CFITSIO\cfortran.h(280,1): error C1189: #error:  "cfortran.h:  Can't find your environment among:    - GNU gcc (g77) on Linux.                                                - MIPS cc and f77 2.0. (e.g. Silicon Graphics, DECstations, ...)         - IBM AIX XL C and FORTRAN Compiler/6000 Version 01.01.0000.0000         - VAX   VMS CC 3.1 and FORTRAN 5.4.                                      - Alpha VMS DEC C 1.3 and DEC FORTRAN 6.0.                               - Alpha OSF DEC C and DEC Fortran for OSF/1 AXP Version 1.2              - Apollo DomainOS 10.2 (sys5.3) with f77 10.7 and cc 6.7.                - CRAY                                                                   - NEC SX-4 SUPER-UX                                                      - CONVEX                                                                 - Sun                                                                    - PowerStation Fortran with Visual C++                                   - HP9000s300/s700/s800 Latest test with: HP-UX A.08.07 A 9000/730        - LynxOS: cc or gcc with f2c.                                            - VAXUltrix: vcc,cc or gcc with f2c. gcc or cc with f77.                 -            f77 with vcc works; but missing link magic for f77 I/O.     -            NO fort. None of gcc, cc or vcc generate required names.    - f2c/g77:   Use #define    f2cFortran, or cc -Df2cFortran               - gfortran:  Use #define    gFortran,   or cc -DgFortran                              (also necessary for g77 with -fno-f2c option)               - NAG f90: Use #define NAGf90Fortran, or cc -DNAGf90Fortran              - Absoft UNIX F77: Use #define AbsoftUNIXFortran or cc -DAbsoftUNIXFortran     - Absoft Pro Fortran: Use #define AbsoftProFortran     - Portland Group Fortran: Use #define pgiFortran     - Intel Fortran: Use #define INTEL_COMPILER" [C:\ROOT-CI\build\CFITSIO-prefix\src\BUILTIN_CFITSIO-build\cfitsio.vcxproj] [C:\ROOT-CI\build\builtins\cfitsio\BUILTIN_CFITSIO.vcxproj]
      (compiling source file '../BUILTIN_CFITSIO/f77_wrap1.c')
```
This PR has been much less trivial than I thought :laughing:

**Comment by bellenot** [comment]
@guitargeek there is no fortran compiler on Windows

**Comment by bellenot** [comment]
> Doesn't have setting the build type the same effect, which is what I do right now?
> 
> ```
>   CMAKE_ARGS -G ${CMAKE_GENERATOR}
>              -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
> ```
> 
> I wonder if we can make the code more platform in-depended like this

As I said, `CMAKE_BUILD_TYPE` is meaningless on Windows

**Comment by bellenot** [comment]
And BTW, I'm working on removing the custom `winrtdebug` flag, but too busy with many other tasks. And then all the Debug builds will fail.

File: builtins/cfitsio/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+             -DCMAKE_C_FLAGS=${CFITSIO_C_FLAGS}
+             -DCMAKE_INSTALL_PREFIX=<INSTALL_DIR>
+             -DBUILD_SHARED_LIBS=OFF
+             -DUSE_CURL=OFF
+             -DZLIB_FOUND=TRUE
+             -DZLIB_INCLUDE_DIR=${ZLIB_INCLUDE_DIRS}
+             -DZLIB_LIBRARIES=$<TARGET_FILE:ZLIB::ZLIB>
+  # Skip the find_package(ZLIB REQUIRED), because we feed CFITSIO our own ZLIB flags.
+  PATCH_COMMAND git apply --ignore-space-change --ignore-whitespace --whitespace=nowarn ${CMAKE_CURRENT_SOURCE_DIR}/cfitsio-no-find-zlib.diff
+  BUILD_COMMAND ${CMAKE_COMMAND} --build ${CFITSIO_BUILD_COMMAND_FLAGS} .
```
Comment (bellenot) [inline builtins/cfitsio/CMakeLists.txt:40]:
```suggestion
  BUILD_COMMAND ${CMAKE_COMMAND} --build . ${CFITSIO_BUILD_COMMAND_FLAGS}
```

**Review by bellenot — CHANGES_REQUESTED** [review]

**Comment by guitargeek** [comment]
I'm making another attempt, just not compiling the fortran wrapper because we don't need it

**Review by bellenot — APPROVED** [review]
Thanks Jonas!

**Pull request merged**